utils/ixdtf/src/parsers/datetime.rs

```diff
@@ -0,0 +1,279 @@
+
+    if dash_two && dash_one {
+        // Advance twice
+        cursor.advance();
+        cursor.advance();
+    } else if dash_two && !dash_one {
+        return Err(ParserError::DateSeparator);
+    }
+
+    let month = parse_date_month(cursor)?;
+
+    cursor.advance_if(cursor.check_or(false, is_hyphen));
+
+    let day = parse_date_day(cursor)?;
+
+    assert_syntax!(cursor.check_or(true, is_annotation_open), InvalidEnd);
+
+    Ok((month, day))
+}
+
+// ==== Unit Parsers ====
+
+#[inline]
+fn parse_date_year(cursor: &mut Cursor) -> ParserResult<i32> {
+    if cursor.check_or(false, is_sign) {
+        let sign = if cursor.next_or(ParserError::ImplAssert)? == '+' {
+            1
+        } else {
+            -1
+        };
+
+        let first = cursor.next_digit()?.ok_or(ParserError::DateExtendedYear)? as i32 * 100_000;
+        let second = cursor.next_digit()?.ok_or(ParserError::DateExtendedYear)? as i32 * 10_000;
+        let third = cursor.next_digit()?.ok_or(ParserError::DateExtendedYear)? as i32 * 1000;
+        let fourth = cursor.next_digit()?.ok_or(ParserError::DateExtendedYear)? as i32 * 100;
+        let fifth = cursor.next_digit()?.ok_or(ParserError::DateExtendedYear)? as i32 * 10;
+
+        let year_value = first
+            + second
+            + third
+            + fourth
+            + fifth
+            + cursor.next_digit()?.ok_or(ParserError::DateExtendedYear)? as i32;
+
+        // 13.30.1 Static Semantics: Early Errors
+        //
+        // It is a Syntax Error if DateYear is "-000000" or "−000000" (U+2212 MINUS SIGN followed by 000000).
+        if sign == -1 && year_value == 0 {
+            return Err(ParserError::DateExtendedYear);
+        }
+
+        let year = sign * year_value;
+
+        return Ok(year);
+    }
+
+    let first = cursor.next_digit()?.ok_or(ParserError::DateYear)? as i32 * 1000;
+    let second = cursor.next_digit()?.ok_or(ParserError::DateYear)? as i32 * 100;
+    let third = cursor.next_digit()?.ok_or(ParserError::DateYear)? as i32 * 10;
+    let year_value =
+        first + second + third + cursor.next_digit()?.ok_or(ParserError::DateYear)? as i32;
+
+    Ok(year_value)
+}
+
+#[inline]
+fn parse_date_month(cursor: &mut Cursor) -> ParserResult<u8> {
+    let first = cursor.next_digit()?.ok_or(ParserError::DateMonth)?;
+    let month_value = first * 10 + cursor.next_digit()?.ok_or(ParserError::DateMonth)?;
+    if !(1..=12).contains(&month_value) {
+        return Err(ParserError::InvalidMonthRange);
+    }
+    Ok(month_value)
+}
+
+#[inline]
+fn parse_date_day(cursor: &mut Cursor) -> ParserResult<u8> {
+    let first = cursor.next_digit()?.ok_or(ParserError::DateDay)?;
+    let day_value = first * 10 + cursor.next_digit()?.ok_or(ParserError::DateDay)?;
+    Ok(day_value)
+}
+
+#[inline]
+fn check_date_validity(year: i32, month: u8, day: u8) -> ParserResult<()> {
+    if !(-271_820..=275_760).contains(&year) {
+        return Err(ParserError::InvalidYearRange);
+    }
```

Comment:
Question: is this an IXDTF limitation or a Temporal limitation?

Reply:
This limit is from the Temporal specification. It can be removed and enforced on the Temporal side